src/components/CourseForm.jsx

```diff
@@ -0,0 +1,76 @@
+import { React, useState} from 'react';
+import { Button, Modal, Form} from 'react-bootstrap';
+
+
+async function addCourse(course){
+    return fetch('url',{
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        method: 'POST',
+        body: JSON.stringify(course)
+    })
+    .then(data=> console.log(data))
+    .catch(err=> console.log(err))
+}
+const StudentForm = () => {
+  const [show, setShow] = useState(false);
+  const [code, setCode] = useState("");
+  const [name, setName] = useState("");
+  const [professor, setProfessor] = useState("");
+  const [program, setProgram] = useState("");
+  const [message, setMessage] = useState("");
```

Comment:
This could get unwieldy for large forms. There are nice libraries out there to help make forms easier
I'm a big fan of Formik https://formik.org/
Can be left for a future iteration